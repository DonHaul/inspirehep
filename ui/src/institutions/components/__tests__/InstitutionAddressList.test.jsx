import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';
import InstitutionAddressList from '../InstitutionAddressList';

describe('InstitutionAddressList', () => {
  it('renders with addresses', () => {
    const addresses = fromJS([
      {
        postal_address: ['Rue Einstein', 'CH-1211 Gen√®ve 23'],
        cities: ['Meyrin'],
        country: 'Switzerland',
        state: 'Geneva',
        place_name: 'CERN',
      },
      {
        cities: ['Geneva'],
        country: 'Switzerland',
      },
    ]);
    const { asFragment } = render(
      <InstitutionAddressList addresses={addresses} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
