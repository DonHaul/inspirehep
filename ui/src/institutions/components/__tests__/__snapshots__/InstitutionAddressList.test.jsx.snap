// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstitutionAddressList renders with addresses 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__ di"
    data-testid="inline-data-list"
  >
    <ul>
      <li>
        <span>
          <div
            class="__InlineList__ di"
          >
            <ul>
              <li>
                <div
                  class="__InlineList__"
                  data-testid="inline-data-list"
                >
                  <ul>
                    <li>
                      Rue Einstein, 
                    </li>
                    <li>
                      CH-1211 Genève 23
                    </li>
                  </ul>
                </div>
                , 
              </li>
              <li>
                <span>
                  Meyrin
                </span>
                , 
              </li>
              <li>
                <span>
                  Switzerland
                </span>
              </li>
            </ul>
          </div>
        </span>
         • 
      </li>
      <li>
        <span>
          <div
            class="__InlineList__ di"
          >
            <ul>
              <li>
                <span>
                  Geneva
                </span>
                , 
              </li>
              <li>
                <span>
                  Switzerland
                </span>
              </li>
            </ul>
          </div>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
